### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by 0xwetzo**
## Description

_Concise description of proposed changes, We recommend using screenshots and videos for better description_

## Additional Information

- [ ] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [ ] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

_Closes #{issue number}_

_Note: If your changes are small and straightforward, you may skip the creation of an issue beforehand and remove this section. However, for medium-to-large changes, it is recommended to have an open issue for discussion and approval prior to submitting a pull request._

Your ENS/address:

### CODE REVIEW

File: packages/hardhat/contracts/Tournament.sol [hunk 1]
```diff
@@ -1,149 +1,797 @@
...
+
+	/////////////////////////////
+	/// Stake & Unstake Funcs ///
+	/////////////////////////////
+
+	/**
+	 * @notice Allows the players to stake their LP token to register for the tournament
+	 */
+	function stakeLPToken() public {
+		require(!isPlayer(msg.sender), "You have already staked");
```
Comment (xrave110) [inline packages/hardhat/contracts/Tournament.sol:190]: Consider of changing to modifier.
More gas efficient is using if statement and revert with some predefined error like:
```
error AlreadyStaked();
..
..
if({condition}
{
revert AlreadyStaked();
}
```

File: packages/hardhat/contracts/Tournament.sol [hunk 2]
```diff
@@ -1,149 +1,797 @@
...
+	/**
+	 * @notice Resolves the game against another player
+	 * @dev Will resolve the game against another player. If against VRF, _stored will be VRF
+	 * @param _senderAddr The address of the player
+	 * @param _senderMove The move the player made
+	 * @param _storedAddr The address of the stored player. Will be 0x0 for VRF
+	 * @param _storedMove The move the stored player made
+	 */
+	function resolveGame(address _senderAddr, uint8 _senderMove, address _storedAddr, uint8 _storedMove) internal {
+		if(_senderMove == _storedMove) {
```
Comment (xrave110) [inline packages/hardhat/contracts/Tournament.sol:407]: Could be just counting the score

File: packages/hardhat/contracts/Tournament.sol [hunk 3]
```diff
@@ -1,149 +1,797 @@
...
+    }
+
+	/**
+	 * @notice Updates the player score by adding points
+	 * @dev Should be called in any case. Also updates player's rank and streak
+	 * @param _player is the address of the player. VRF will be 0x0
+	 * @param _points 0 = lost, 1 = draw, 2 = won
+	 */
+	function updateScore(address _player, uint8 _points) internal {
+		if(_player == address(0)) return; // We don't update VRF score
```
Comment (xrave110) [inline packages/hardhat/contracts/Tournament.sol:434]: Most code could be done in a front end

File: packages/hardhat/contracts/Tournament.sol (hunk 1 above)
Comment (0xwetzo) [reply packages/hardhat/contracts/Tournament.sol:190]: Ok, good to know

**Pull request merged**